common/src/main/java/com/microsoft/identity/common/internal/cache/ADALOAuth2TokenCache.java

```diff
@@ -20,10 +34,22 @@
 
     SharedPreferencesFileManager mSharedPreferencesFileManager;
     final static String SHARED_PREFERENCES_FILENAME = "com.microsoft.aad.adal.cache";
+    private static final String TAG = "ADALOAuth2TokenCache";
+    @SuppressLint("StaticFieldLeak")
+    private static StorageHelper sHelper;
```

Comment:
Does this need to be `static`? If this object gets initialized with an `Activity` as the `Context` it could potentially cause some nasty memory leaks

Reply:
This is code that i ported from the existing implementation... i'll review and see if we can make this non-static... i suspect it's static to serialize access to the cache and I'm not sure that's actually required.

Reply:
I updated the code to ensure that an application context is being used for now.  I think we should do this in ADAL as well.

Reply:
Yep - I'll file an issue for this and link-back here

Reply:
Issue filed: https://github.com/AzureAD/azure-activedirectory-library-for-android/issues/1056